### PR TITLE
Various small fixes - part 11

### PR DESCRIPTION
**Pull request by DigitalBrains1**
Various small fixes

Please only merge just before releasing 1.6.3 so we can still amend this with further small fixes we find in the meantime. Please squash-merge so we don't get tiny commits in our repo.

Improve `Show BlackBox`: add parentheses for proper precedence for `BBTemplate`. It showed in `show (BlackBoxE ... (BBTemplate ...) ...)`, where the parentheses were missing, making it a surprising read.

## Still TODO:

  - [ ] Write a changelog entry (see changelog/README.md)
  - [ ] Check copyright notices are up to date in edited files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ git checkout 1.2
 
 Note that release branches might contain non-released patches.
 
-## GHC compatability
+## GHC compatibility
```
Comment (martijnbastiaan) [inline README.md:65]: oof

**Comment by martijnbastiaan** [comment]
Looks good, let's merge?

**Comment by leonschoorl** [comment]
Oh no, I found more: https://github.com/clash-lang/clash-compiler/commit/0d2eda7b5627a14332ab49ec86e8b53ddea0b105

**Comment by DigitalBrains1** [comment]
Things worked out differently than we anticipated. We really _were_ about to release 1.6.3, but then you mentioned you had almost fixed the `fromInteger#` issue, and we decided we should wait for that. Those two bits of information literally missed each other by minutes. Such is life.

By the way, did you cross-check the tutorial now exhaustively lists all holes?

**Comment by leonschoorl** [comment]
@DigitalBrains1 I have now: #2161

**Pull request merged**